File: flutter/lib/features/recipes/domain/recipe_category.dart
class RecipeCategory {
  final int id;
  final int position;
  final String name;
  final String color;
  final String icon;

  const RecipeCategory({
    required this.id,
    required this.position,
    required this.name,
    this.color = '#0052CC',
    this.icon = '🍽',
  });

  factory RecipeCategory.fromJson(Map<String, dynamic> json) {
    return RecipeCategory(
      id: json['id'] as int,
      position: json['position'] as int? ?? 0,
      name: json['name'] as String,
      color: json['color'] as String? ?? '#0052CC',
      icon: json['icon'] as String? ?? '🍽',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'color': color,
      'icon': icon,
      if (position != 0) 'position': position,
    };
  }
}

File: backend/app/models/recipe_category.py
from sqlalchemy import ForeignKey, String, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column

from ..database import Base


class RecipeCategory(Base):
    __tablename__ = "recipe_categories"
    __table_args__ = (
        UniqueConstraint("family_id", "name", name="uq_recipe_category_family_name"),
    )

    id: Mapped[int] = mapped_column(primary_key=True)
    family_id: Mapped[int] = mapped_column(
        ForeignKey("families.id", ondelete="CASCADE"), index=True
    )
    position: Mapped[int] = mapped_column(default=0, index=True)
    name: Mapped[str] = mapped_column(String(50))
    color: Mapped[str] = mapped_column(String(7), default="#0052CC")
    icon: Mapped[str] = mapped_column(String(10), default="🍽")
